--- a/src/function/scalar/string/length.cpp
+++ b/src/function/scalar/string/length.cpp
@@ -266,6 +266,9 @@ ScalarFunctionSet OctetLengthFun::GetFunctions() {
 	// length for BLOB type
 	ScalarFunctionSet octet_length("octet_length");
 	octet_length.AddFunction(ScalarFunction({LogicalType::BLOB}, LogicalType::BIGINT,
+	                                        ScalarFunction::UnaryFunction<string_t, int64_t, StrLenOperator>));
+	// MariaDB's SELECT_LEX::print() rewrites LENGTH() to octet_length(); support VARCHAR too
+	octet_length.AddFunction(ScalarFunction({LogicalType::VARCHAR}, LogicalType::BIGINT,
 	                                        ScalarFunction::UnaryFunction<string_t, int64_t, StrLenOperator>));
 	octet_length.AddFunction(ScalarFunction({LogicalType::BIT}, LogicalType::BIGINT,
 	                                        ScalarFunction::UnaryFunction<string_t, int64_t, OctetLenOperator>));
